nightfall-optimist/src/services/check-block.mjs

```diff
@@ -218,7 +218,12 @@ export async function checkBlock(block, transactions) {
   try {
     for (let i = 0; i < transactions.length; i++) {
       transaction = transactions[i];
-      await checkTransaction(transaction, false, { blockNumberL2: block.blockNumberL2 }); // eslint-disable-line no-await-in-loop
+      // eslint-disable-next-line no-await-in-loop
+      await checkTransaction(
```

Comment:
Why do we need to check transactions from blocks that we have proposed? If we have done checks during block assembly, why do we need to do them again?

Reply:
@druiz0992 
same as above comment - case when proposer is bad-proposer

> I think intention is to keep check at the time receive is because, consider a case where proposer is a bad proposer who create a bad-block(overriding the code of check of block assembly - for example) we still need transaction check at receive event-handler in optimist. also if you notice once check fails only we trigger create challenge (is challenger exist in optimist)